esp32: is broken for the latest esp-idf version

Following the ESP32's instructions for installing and using the espressif SDK resulted in a broken build.

Specifically, I get the below error message when building hello_world.

Git blame shows that the offending line was added in mid-august, so I am guessing that the latest ESP-IDF SDK is not compatible with Zephyr. Haven't confirmed this by going back in time yet though ...

## Solution suggestions

1. Put a specific git revision in the SDK installation instructions.
2. Fix the latest ESP-IDF version to be compatible with Zephyr and then set up CI to ensure the latest ESP-IDF is always compatible with Zephyr.

## Make output

`  /home/sebo/workspace_zephyr/esp/xtensa-esp32-elf/bin/xtensa-esp32-elf-gcc -Wp,-MD,drivers/gpio/.gpio_esp32.o.d  -nostdinc -isystem /home/sebo/workspace_zephyr/esp/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/5.2.0/include -isystem /home/sebo/workspace_zephyr/esp/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/5.2.0/include-fixed -I/home/sebo/zephyr/kernel/include -I/home/sebo/zephyr/arch/xtensa/include -I/home/sebo/zephyr/arch/xtensa/soc/esp32 -I/home/sebo/zephyr/boards/xtensa/esp32  -I/home/sebo/zephyr/include -I/home/sebo/zephyr/include -I/home/sebo/workspace_zephyr/zephyr/samples/hello_world/outdir/esp32/include/generated -include /home/sebo/workspace_zephyr/zephyr/samples/hello_world/outdir/esp32/include/generated/autoconf.h  -I/home/sebo/workspace_zephyr/esp/esp-idf/components/esp32/include -I/home/sebo/workspace_zephyr/esp/esp-idf/components/soc/esp32/include -I/home/sebo/zephyr/lib/libc/minimal/include  -I/home/sebo/zephyr/drivers/gpio -Idrivers/gpio -DKERNEL -D__ZEPHYR__=1 -c -g -std=c99 -Wall -Wformat -Wformat-security -D_FORTIFY_SOURCE=2 -Wno-format-zero-length -Wno-main -ffreestanding -Os -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-stack-protector -ffunction-sections -fdata-sections -fms-extensions -Wno-unused-but-set-variable -fno-reorder-functions -fno-defer-pop -Wno-pointer-sign -fno-strict-overflow -Werror=implicit-int   -I/home/sebo/zephyr/include/drivers   -I/home/sebo/zephyr/drivers    -D"KBUILD_STR(s)=#s" -D"KBUILD_BASENAME=KBUILD_STR(gpio_esp32)"  -D"KBUILD_MODNAME=KBUILD_STR(gpio_esp32)" -c -o drivers/gpio/gpio_esp32.o /home/sebo/zephyr/drivers/gpio/gpio_esp32.c
In file included from /home/sebo/workspace_zephyr/esp/esp-idf/components/soc/esp32/include/soc/dport_access.h:20:0,
                 from /home/sebo/workspace_zephyr/esp/esp-idf/components/soc/esp32/include/soc/dport_reg.h:20,
                 from /home/sebo/zephyr/drivers/gpio/gpio_esp32.c:9:
/home/sebo/workspace_zephyr/esp/esp-idf/components/esp32/include/esp_dport_access.h:15:23: fatal error: sdkconfig.h: No such file or directory
`

--- a/boards/xtensa/esp32/doc/esp32.rst
+++ b/boards/xtensa/esp32/doc/esp32.rst
@@ -62,7 +62,6 @@ The toolchain is available for Linux, Windows, and Mac hosts and
 instructions to obtain and set them up are available in the ESP-IDF
 repository, using the toolchain and SDK links above.
 
-
 Set up build environment
 ========================
 
@@ -78,6 +77,17 @@ located, and where the toolchain has been installed:
    export ESP_IDF_PATH="/path/to/esp-idf"
    export ESPRESSIF_TOOLCHAIN_PATH="/path/to/xtensa-esp32-elf/"
 
+Since ESP-IDF is an external project in constant development, it's possible
+that files that Zephyr depends on will be moved, removed, or renamed.  Those
+files are mostly header files containing hardware definitions, which are
+unlikely to change and require fixes from the vendor.  In addition to
+setting the environment variables above, also check out an earlier version
+of ESP-IDF:
+
+.. code-block:: console
+
+   cd ${ESP_IDF_PATH}
+   git checkout dc8c33892e0
 
 Flashing
 ========

